harness/determined/estimator/_estimator_trial.py

```diff
@@ -660,11 +664,33 @@ def average_metrics(self, metrics: Dict[str, Any]) -> Optional[Dict[str, Any]]:
                         metrics[metric_name] += worker_metric[metric_name] / hvd.size()
             return metrics
         else:
-            self.train_process_comm_worker = cast(ipc.ZMQClient, self.train_process_comm_worker)
+            self.train_process_comm_worker = cast(
+                ipc.ZMQBroadcastClient, self.train_process_comm_worker
+            )
             logging.debug(f"Worker {hvd.rank()} sending metrics.")
-            self.train_process_comm_worker.barrier(message=metrics)
+            self.train_process_comm_worker.send(metrics)
             return None
 
+    def allgather_metrics(self, stuff: Any) -> List:
```

Comment:
non-blocking: rename `stuff` to `data`, `metrics` or  something of the ilk

Reply:
fixed